loopchain/channel/channel_service.py

```diff
@@ -497,20 +497,90 @@ def connect_to_radio_station(self, is_reconnect=False):
+
+        if response['response_code'] == message_code.Response.success:
+            if TimerService.TIMER_KEY_SUBSCRIBE in self.__timer_service.timer_list.keys():
+                self.__timer_service.stop_timer(TimerService.TIMER_KEY_SUBSCRIBE)
+                logging.debug(f"Subscription to RadioStation(mother peer) is successful.")
+
+            if TimerService.TIMER_KEY_SHUTDOWN_WHEN_FAIL_SUBSCRIBE in self.__timer_service.timer_list.keys():
+                self.__timer_service.stop_timer(TimerService.TIMER_KEY_SHUTDOWN_WHEN_FAIL_SUBSCRIBE)
+
+            # start next get_status timer
+            timer_key = TimerService.TIMER_KEY_GET_LAST_BLOCK_KEEP_CITIZEN_SUBSCRIPTION
+            if timer_key not in self.__timer_service.timer_list.keys():
+                util.logger.spam(f"add timer for check_block_height_call to radiostation...")
+                self.__timer_service.add_timer(
+                    timer_key,
+                    Timer(
+                        target=timer_key,
+                        duration=conf.GET_LAST_BLOCK_TIMER,
+                        is_repeat=True,
+                        callback=self.__check_block_height_call_to_rs_stub,
+                        callback_kwargs={"rs_rest_stub": rs_rest_stub}
+                    )
+                )
         else:
-            peer_target = f"https://{ChannelProperty().rest_target}"
-        response = rs_rest_stub.call(
-            "Subscribe", {
-                'channel': ChannelProperty().name,
-                'peer_target': peer_target
-            }
-        )
-        if response['response_code'] != message_code.Response.success:
-            error = f"subscribe fail to peer_target({ChannelProperty().radio_station_target}) " \
-                    f"reason({response['message']})"
-            await StubCollection().peer_stub.async_task().stop(message=error)
+            timer_key = TimerService.TIMER_KEY_SHUTDOWN_WHEN_FAIL_SUBSCRIBE
+            if timer_key not in self.__timer_service.timer_list.keys():
+                error = f"Shutdown by Subscribe retry timeout({conf.SHUTDOWN_TIMER})"
+                self.__timer_service.add_timer(
+                    timer_key,
+                    Timer(
+                        target=timer_key,
+                        duration=conf.SHUTDOWN_TIMER,
+                        callback=self.shutdown_peer,
+                        callback_kwargs={"message": error}
+                    )
+                )
+
+        return response
+
+    def __check_block_height_call_to_rs_stub(self, **kwargs):
+        rs_rest_stub = kwargs.get("rs_rest_stub", None)
+        try:
+            last_block = rs_rest_stub.call("GetLastBlock")
+            if last_block['height'] <= self.__block_manager.get_blockchain().block_height:
+                return
+            else:
+                # citizen needs additional block or connected to problematic peer.
+                raise Exception
```

Comment:
This exception does not seem useful.
Why don't you remove `try, except` and `else`